bgpd/bgp_route.c

```diff
@@ -10184,6 +10203,32 @@ static int bgp_show_neighbor_route(struct vty *vty, struct peer *peer,
 	return bgp_show(vty, peer->bgp, afi, safi, type, &peer->su, use_json);
 }
 
+DEFUN (show_ip_bgp_flowspec_routes_detailed,
+       show_ip_bgp_flowspec_routes_detailed_cmd,
+       "show [ip] bgp [<view|vrf> VIEWVRFNAME] ["BGP_AFI_CMD_STR" flowspec] detail [json]",
+       SHOW_STR
+       IP_STR
+       BGP_STR
+       BGP_INSTANCE_HELP_STR
+       BGP_AFI_HELP_STR
+       "SAFI Flowspec\n"
+       "Detailed information on flowspec entries\n"
+       JSON_STR)
+{
+	afi_t afi = AFI_IP;
+	safi_t safi = SAFI_UNICAST;
+	struct bgp *bgp = NULL;
+	int idx = 0;
+
+	bgp_vty_find_and_parse_afi_safi_bgp(vty, argv, argc, &idx, &afi, &safi,
+					    &bgp);
+	if (!idx)
+		return CMD_WARNING;
+
+	return bgp_show(vty, bgp, afi, safi,
+			bgp_show_type_detail, NULL, use_json(argc, argv));
+}
```

Comment:
@pguibert6WIND can you show an example output of a show route table, show a specific route and a show flowspec detail?  Why not just show detail when a specific prefix is provided in the show command? I think this would be more consistent with other safis

Reply:
Hi @louberger ,
following link can be helpful.
https://gist.github.com/pguibert6WIND/e2ba176d04f8252277ecdd8ce419afd9

This is a real challenge to display correctly all the kind of FS NLRI.
for now, I propose 3 kinds of output: detailed, summarised ( on a line with some commas for separating ), and concatenated ( no commas).

I am open to proposals.

Note also that I plan to do a show policy-route that will be a conversion from the original NLRI.
Actually, you can have several policy entries for a NLRI ( FS supports OR rules, so it extends the combinatories).

The show implementation right now, is , from my perspective, basic.
And there will probably be enhancements.

Reply:
@pguibert6WIND thank you for the output samples.  This is looks good to me,

Reply:
Do we care about column alignment in the non detail show?  Should we even bother with a non-detailed show?  Does showing only a small portion of the flowspec rule make sense in the context of needing to show all of it?

Reply:
currently, the columns alignment is not handled.
I understand that you wonder if the non detail show is really necessary? 

actually, I looked at how vendors do. cisco provides that kind of show non detailed.
removing it is not appropriate solution.

second point, even if you can do many things with FS, I bet on the realistic view of operators to inject consistent and rather simple entries. I don't think operator wants to use such complexity.

third point: mitigation point for column alignemtn.
I think of returning a multiple line detail, aligned with a certain number of bytes after startup.
This could improve readability, and would not shift the other elements.

What is discussed here is out of scope of the proposed patch.



show bgp ipv4 flowspec

prefix                           NH         ...
from @A to @B          |              ....
   port [55;200], tcp 43|
   proto tcp                  |